client/verta/verta/dataset/_s3.py

```diff
@@ -97,10 +97,16 @@ def _get_s3_loc_metadata(cls, s3_loc):
         if s3_loc.key is None:
             # TODO: handle `bucket_name` not found
             for obj in s3.list_object_versions(Bucket=s3_loc.bucket)['Versions']:
-                if obj['IsLatest']:
-                    yield cls._get_s3_obj_metadata(obj, s3_loc.bucket, obj['Key'])
+                if obj['Key'].endswith('/'):  # folder, not object
+                    continue
+                if not obj['IsLatest']:
+                    continue
+                yield cls._get_s3_obj_metadata(obj, s3_loc.bucket, obj['Key'])
         else:
             # TODO: handle `key` not found
+            if s3_loc.key.endswith('/'):
+                s3_path = cls._S3_PATH.format(s3_loc.bucket, s3_loc.key)
+                raise ValueError("{} must be a bucket or an object, not a folder".format(s3_path))
```

Comment:
Why wouldn't we allow a folder and then we scrape it?

Reply:
Because that's a much more complex change that I think is outside the scope of this ticket.

I can make a new ticket to scrape folders.

Reply:
Actually, totally doable—took 20 minutes.